substrate/frame/revive/src/vm/evm/instructions/control.rs

```diff
@@ -137,6 +137,8 @@ pub fn stop<'ext, E: Ext>(context: Context<'_, 'ext, E>) {
 
 /// Invalid opcode. This opcode halts the execution.
 pub fn invalid<'ext, E: Ext>(context: Context<'_, 'ext, E>) {
+	log::info!("INVALID");
+	context.interpreter.extend.gas_meter_mut().consume_all();
```

Comment:
why do we need that?

Reply:
I guess they must be matching somewhere on the halt reason and remove the remaining gas

Reply:
yeah they do
Self::Return(InterpreterResult::new(InvalidFEOpcode, Bytes::new(), gas))

ok this is fine then